modules/transcription-service-amberscript/src/main/java/org/opencastproject/transcription/amberscript/AmberscriptTranscriptionService.java

```diff
@@ -273,11 +320,32 @@ public Job startTranscription(String mpId, Track track) throws TranscriptionServ
 
   @Override
   public Job startTranscription(String mpId, Track track, String... args) throws TranscriptionServiceException {
-    String language;
-    if (args.length > 0 && StringUtils.isNotBlank(args[0])) {
-      language = args[0];
-    } else {
-      language = getLanguage();
+    String language = null;
+
+    if (languageFromDublinCore) {
+      for (Catalog catalog : track.getMediaPackage().getCatalogs(MediaPackageElements.EPISODE)) {
+        if (language != null) {
```

Comment:
Why not move the `break` down to after the (hopefully) successful language fetch?  Ie, stick it on line 336, and skip re-looping.